src/custom_types/color_picker.py

```diff
@@ -0,0 +1,234 @@
+import functools
+import json
+import string
+from asyncio import Future
+
+from prompt_toolkit.application.current import get_app
+from prompt_toolkit.buffer import Buffer
+from prompt_toolkit.layout import ScrollablePane
+from prompt_toolkit.layout.containers import HSplit
+from prompt_toolkit.layout.dimension import D
+from prompt_toolkit.styles import Style
+from prompt_toolkit.widgets import Button, Dialog, Frame, Label, TextArea
+
+from constants import USER_SETTINGS_DIR
+from custom_types.ui_types import PopUpDialog
+
+
+class ColorPicker(PopUpDialog):
+    """Text Input for the open dialog box"""
+
+    def __init__(self, style_class: str, style_class_attr: str):
+        self.future = Future()
+        with open(USER_SETTINGS_DIR, "r") as user_file:
+            self.user_settings = json.load(user_file)
+
+        def is_hex(s: str) -> bool:
+            """Validate that the user entered text is a 6 digit hex number
+
+            1. len(6)
+            2. only contains 0-9 and a-f
+            """
+            if len(s) == 6:
+                hex_digits = set(string.hexdigits)
+                # if s is long, then it is faster to check against a set
+                return all(c in hex_digits for c in s)
+            else:
+                return False
+
+        def string_to_dict(s: str) -> dict:
+            """Turns a single style classes string into a dict
+
+            Ex 'bg:#123456 fg:#654321 #abcdef' -> {'bg':123456, 'fg':'654321', '':'abcdef'}
+            """
+            out = {}
+            if len(s) == 0:
+                return out
+            items = s.split(" ")
+            for elm in items:
+                key, pair = elm.split("#")
+                out[key.replace(":", "")] = "#" + pair
```

Comment:
```suggestion
                key, pair = elm.split(":")
                out[key] = pair
```

Reply:
Ignore this if you actually want an empty key name but I think it could probably be renamed to "text" since the 3 options are foreground, background, text